crates/cargo-util/src/process_builder.rs

```diff
@@ -41,6 +41,11 @@ pub struct ProcessBuilder {
     retry_with_argfile: bool,
     /// Data to write to stdin.
     stdin: Option<Vec<u8>>,
+    /// Indicates whether a failure of this process should be considered
+    /// spurious. This allows distinguishing between errors that are
+    /// expected to sometimes fail (e.g., flaky tools) and those that
+    /// are real failures.
+    failure_is_spurious: bool,
```

Comment:
`ProcessBuilder` is a general Command wrapper and used in a couple of different places. It feels a bit odd we extend this there. What about we just creates a GitCliError to wrap ProcessError in `fetch_with_cli`, and in retry check that error?